sdk/core/Azure.Core/src/Shared/TaskExtensions.cs

```diff
@@ -87,5 +91,39 @@ private static void VerifyTaskCompleted(bool isCompleted)
                 throw new InvalidOperationException("Task is not completed");
             }
         }
+
+        public readonly struct Enumerable<T> : IEnumerable<T>
+        {
+            private readonly IAsyncEnumerable<T> _asyncEnumerable;
+
+            public Enumerable(IAsyncEnumerable<T> asyncEnumerable) => _asyncEnumerable = asyncEnumerable;
+
+            public Enumerator<T> GetEnumerator() => new Enumerator<T>(_asyncEnumerable.GetAsyncEnumerator());
+
+            IEnumerator<T> IEnumerable<T>.GetEnumerator() => new Enumerator<T>(_asyncEnumerable.GetAsyncEnumerator());
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
+        public readonly struct Enumerator<T> : IEnumerator<T>
+        {
+            private readonly IAsyncEnumerator<T> _asyncEnumerator;
+
+            public Enumerator(IAsyncEnumerator<T> asyncEnumerator) => _asyncEnumerator = asyncEnumerator;
+
+#pragma warning disable AZC0107 // DO NOT call public asynchronous method in synchronous scope.
```

Comment:
(extreme) nit:  All other uses of `Do not` are normal-cased.   These are YELLING!   Should the case be adjusted for these comments?

Reply:
Fixed